File: back/src/main/java/com/speakerspace/model/session/Session.java
package com.speakerspace.model.session;

import com.google.cloud.spring.data.firestore.Document;
import jakarta.validation.constraints.NotBlank;
import lombok.*;

import java.util.Date;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Builder
@Document
public class Session {
    @NotBlank(message = "ID is required")
    @EqualsAndHashCode.Include
    private String id;
    private Date start;
    private Date end;
    private String track;
    private String title;
    private String abstractText;
    private String deliberationStatus;
    private String confirmationStatus;
    private String level;
    private String references;
    private List<Format> formats;
    private List<Category> categories;
    private List<String> tags;
    private List<String> languages;
    private List<String> speakerIds;
    private Reviews reviews;
    private String eventId;
    private Date createdAt;
    private Date updatedAt;

    public Session() {
        Date now = new Date();
        this.createdAt = now;
        this.updatedAt = now;
    }
}

File: front/src/app/feature/admin-management/type/session/session.ts
export type SessionImportData = {
  id?: string;
  title: string;
  abstractText: string;
  deliberationStatus: string;
  confirmationStatus: string;
  level: string;
  references?: string;
  formats: Format[];
  categories: Category[];
  tags: string[];
  languages: string[];
  speakers: Speaker[];
  reviews: Reviews | null;
  eventId?: string;
  createdAt?: string;
  updatedAt?: string;
  start?: Date;
  end?: Date;
  track?: string;
}

export type SessionImportRequest = {
  eventId: string;
  sessions: SessionImportData[];
}

export type ImportResult = {
  successCount: number;
  totalCount: number;
  errors: string[];
}

export type Format = {
  id: string;
  name: string;
  description: string;
}

export type Category = {
  id: string;
  name: string;
  description: string;
}

export type Speaker = {
  id: string;
  name: string;
  bio: string;
  company: string;
  references: string;
  picture: string;
  location: string;
  email: string;
  socialLinks: string[];
}

export type Reviews = {
  average: number;
  positives: number;
  negatives: number;
}
